cloudbuild.yaml

```diff
@@ -323,7 +323,7 @@ steps:
           featureWithoutGate=""
           testClusterLocation="${region}"
           { stdbuf -oL -eL gcloud builds submit . --suppress-logs --config=./ci/e2e-test-cloudbuild.yaml \
-            --substitutions _FEATURE_WITH_GATE=$featureWithGate,_FEATURE_WITHOUT_GATE=$featureWithoutGate,_CLOUD_PRODUCT=$cloudProduct,_TEST_CLUSTER_NAME=$testCluster,_TEST_CLUSTER_LOCATION=$testClusterLocation,_REGISTRY=${_REGISTRY},_PARENT_COMMIT_SHA=${COMMIT_SHA},_PARENT_BUILD_ID=${BUILD_ID} \
+            --substitutions _FEATURE_WITH_GATE=$featureWithGate,_CLOUD_PRODUCT=$cloudProduct,_TEST_CLUSTER_NAME=$testCluster,_TEST_CLUSTER_LOCATION=$testClusterLocation,_REGISTRY=${_REGISTRY},_PARENT_COMMIT_SHA=${COMMIT_SHA},_PARENT_BUILD_ID=${BUILD_ID} \
```

Comment:
Unless I'm misreading them, the two places we set `featureWithGate` are now the same and can be consolidated into a single variable (e.g. moved outside of the if statement). 

The only thing that would remain inside of the `if [ $cloudProduct = generic ]` block is how we pick the name of the test cluster.

Reply:
I think `featureWithGate` are still different. For Generic, `SplitControllerAndExtensions` is set to false while in Autopilot it's set to true

Reply:
I think we can set both to be true now that its in beta?

Reply:
oh that makes sense. We can set both to true.

Reply:
Looking at https://github.com/googleforgames/agones/pull/3165 again, I think we may need to wait until we hit stable to consolidate the variables. That PR flipped the `featureWithGate` for non-autopilot to false whereas on autopilot it needs to stay true.